### ISSUE TITLE
CodeMaid crashes Visual Studio when reorganizing file during rename

### ISSUE THREAD

**Issue by mstromb** [codecadwallader/codemaid#142]
Right-click a file in solution explorer -> Rename -> Choose to rename references

CodeMaid tries to reorganize the renamed file until it crashes VS

I can reproduce this consistently on VS2013u5 with CodeMaid v.0.8.1 from the gallery

**Comment by codecadwallader** [comment]
Can you share your configuration?  It sounds like at a minimum you have enabled automatic cleanup on save, and automatic reorganization on cleanup.

**Comment by mstromb** [comment]
Yes, those are both enabled. I think I have mostly defaults set otherwise. Here's a gist with the project and user settings I exported: https://gist.github.com/mstromb/0637c77b2e4e940078f1

**Comment by mstromb** [comment]
I should add that you don't need anything interesting in the file to trigger this. Adding a new .cs file to a project and then immediately renaming it is sufficient to trigger the bug. Clumsy typing is what led me to find this one :)

**Comment by codecadwallader** [comment]
Thanks for providing your configuration.  I was able to reproduce what you are seeing only once I had disabled "Use undo transactions" -> "Skip during automatic cleanup on save".  There are three different configuration locations where we have "Skip during automatic cleanup on save" sub-options to bypass Visual Studio functionality that doesn't play nicely during automated scenarios.

If you re-enable that option, the scenario works as expected for me.  Does that help?

**Comment by mstromb** [comment]
Yes, when undo transactions are off it works fine.

I had them on because it seemed that having auto-format on and saving a file that was missing a curly brace would basically turn the file into gibberish, and then I'd have to spend 30s holding down ctrl-z to undo 200 line changes. Now I can't seem to reproduce that behavior when I try, so I can just leave skip undo enabled.

**Comment by codecadwallader** [comment]
Sounds good, closing the issue.

**Comment by glen-nicol** [comment]
@mstromb  I ran into this exact scenario. Holding undo was really annoying so I turned that option off then rename crashes visual studio.

Specifically, I had typed the something like following and hit ctrl+s and my file was completely broken afterwards.

```charp
private static Random = new Random()
```

Basically didn't put an identifier for the field and I think I left the semicolon off.

I'm using VS 2013.
